README.md

```diff
@@ -331,6 +333,21 @@ Specific id for container. The container renders with a default id of `rdtsN` wh
 
 Use to ensure a own unique id when a simple counter is not sufficient, e.g in a partial server render (SSR)
 
+### label
+
+Type: `string`
+
+Adds `aria-labelledby` to search input when input starts with `#`, adds `aria-label` to search input when label has value (not containing '#')
+
+### enableKeyboardNavigation
+
+Type: `bool` (default: `false`)
+
+Adds navigation with `arrow` keys, `page down/up` / `home/end` and toggle of selection with `enter`. `Arrow/page up/down` also toggles open of dropdown if closed.
+
+To close open dropdown `escape` or `tab` can be used and `backspace` can be used for deletion of tags on empty search input.
```

Comment:
`tab` for closing as in tabbing out? I'm Ok with just `esc` for closing which is what the native controls do.

Reply:
Well if you've tabbed in you probably will tab out. But perhaps one should select the current "focused" item also then? But that should probably only happen for simpleSelect.

Tab into this example, use arrow down and tab out. It selects the current focused item.
https://www.w3.org/TR/wai-aria-practices/examples/listbox/listbox-collapsible.html

Reply:
For simple select - yes.

For multi, tabbing becomes tricky - if I focus on the input and tab, am I tabbing out or tabbing to next tag? Tabs cycle through controls so one could argue that it should go to next tag and then to its `x` button as well. But then, we can probably skip cycling to `x` if one can `tab > select tag > del > delete tag`. 

Maybe we should first write up a spec of what the keyboard navigation should be like in different cases - it'll help us validate our code as well.

Eventually, it can go in the Readme as keyboard map help section.

Reply:
We [accomplished this](https://github.com/dowjones/react-dropdown-tree-select/pull/233#discussion_r271608037) yesterday in #233 I think. It does not seem to be pushed yet though.
We added a keydownevent on the now "tabable" trigger-a. This and the click event now triggers handleClick as before. But we added a callback checking if the dropdown is visible after the state change and then sets focus on the input-field in this case. This makes it possible to tab into the dropdown. Clicking enter, space or arrowdown you get focus on the search input and the dropdown is visible. Tabbing takes you to the next delete-button if any. After the tag-deletes you get focus on the input if tabbing further.

Reply:
Now it's in place you can see the combined result here. https://ellinge.github.io/react-dropdown-tree-select-test/#DevelopTemp-default

Reply:
Nice! I see that tabbing between input and tags work beautifully. But neither backspace not delete had any effect on the tags. Tabbing also doesn't go to the `x` button but I think that's ok if backspace and/or delete can delete those. 

On the other hand, `tag > enter > delete` doesn't feel right. Enter is never associated with delete unless the focus is explicitly on a delete button (in our case the `x` button). Which is another reason I think `x` should not be tab-able - it just feels like too many tabs to navigate.

Reply:
It does go to the x-button but I've highligheted the current tab also. So not sure what to make of your last paragraph.

Reply:
I added keydown for backspace and delete now also when focusing on a tag btw.
The test page is updated. See if it stills feels wrong?

Reply:
Moved out keyboard nav to a separate section in the readme also. Not sure how detailed it should be and still be clear.

Reply:
> Not sure how detailed it should be and still be clear.

We won't need a detailed section if the navigation is natural and intuitive :smile:

Some observations (and by the way - I love your example page - makes it so easy to test out the changes):

 - If I hit space here, it goes into search mode. We're toggling the checkboxes with but usually, space toggles them for native controls also. Is this a focus issue? I think if the label or checkbox has the focus, then space would work but for some reason, input has the focus.
![image](https://user-images.githubusercontent.com/781818/55651639-49c83500-57b7-11e9-9817-ec90ed53b30f.png)

 - `tab > tag > delete > focus gone`. One needs to tab again to select the next one or enter the search box. Will it be too complex to move the focus forward - to whatever is next (select next tag if present else move the focus to search box)? 

 - There was also another case which I can't seem to reproduce now - what happend was I had the dropdown open, and it stayed open somehow when I moved to next one. I shift tabbed to move to previous and it first went to firt dropdown item and then shift tab moved to the search box and so on. So shift tabbing worked but it felt odd (even though it was the right order). I expected it to tab to previous control (and not a dropdown item). I can't reproduce it now though.

Reply:
Speaking of Space key toggling checkbox, are we currently handling the `partialSelect` use case? A parent level checkbox can have a partial selected state and space will need to cycle through three states. Will need `aria-checked="mixed"` attributes as well during partial state.

Reply:
Well space is also part of the possible search entry. So normally yes. You would select with space. But not when you have a input query. See:
https://www.w3.org/TR/wai-aria-practices-1.1/examples/combobox/aria1.1pattern/listbox-combo.html

Reply:
Added focus after delete-logic now. Added it in onTagRemove and had to use getElementById to find the next target. Not sure how we would pass refs etc up but that would be an option as well.

Reply:
I'll look into handling tabbing better also. For simpleSelect. Toggle check (only select, not deselect) on tab.
For the other we either do it like now, just leave, or we handle it like next prev node. One option would be to tab to any action in this case also.

I also need to fix shift-tab to go to previous tag if any. Just not sure how we should handle it if tabbing also navigates tree. Escape and shift-tab? Would be simpler if we just leaved.

Reply:
I've only fixed so tab also selects current focused node on simpleSelect now. Had some issues in firefox with clicking input field and shift-tab. Did not go to lasts tag.